src_cpp/union_find.hpp

```diff
@@ -557,7 +653,7 @@ class UfDecoder{
 
                 invalid_clusters.clear();
                 for(auto cl: clusters){
-                    if(cl->active == true && cl->parity() == 1){
+                    if(cl->active == true && cl->parity() == 1 && cl->contains_boundary_bits == false){
```

Comment:
So you turn off a cluster whenever it grows into a boundary. Are there cases and UF grow orders for which this is suboptimal (or better) than growing until it becomes valid, i.e., merging with another invalid cluster? I am almost certain there are multiple cases where the resulting clusters are not equivalent.

Reply:
As soon as a cluster intersects a boundary, I mark it as "valid". I.e. it no longer grows. However, other clusters can grow into it and merge.

Reply:
As for whether this is optimal ... probably not. I've simply decided to treat boundaries as equivalent to defects in the lattice. Hence the decision to freeze growth once the boundary is hit.

Reply:
Thinking about this some more, shouldn't it generally be the case that we want to keep our clusters as small as possible.?

Reply:
Thinking about it more, I am not sure it makes a difference, yes.